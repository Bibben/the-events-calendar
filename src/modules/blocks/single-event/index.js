/**
 * External Dependencies
 */
const { __ } = wp.i18n;
const { useBlockProps } = wp.blockEditor;

/**
 * Small component to simplify some pseudo event blocks.
 *
 * @param width
 * @returns {JSX.Element}
 * @constructor
 */
const FauxLine = ( {style ={}, ...props } ) => {
    const divStyle = { height: 16, background: '#eee', margin: '18px 4px 18px 0' }

    return ( <div style={{...divStyle, ...style}  } {...props} />    )
}

/**
 * The Archive Events block used in Site Editor templates.
 */
export default {
    id: 'tec/single-event',
    title: __( 'Single Event', 'the-events-calendar' ),
    icon: 'calendar-alt',
    category: 'tribe-events',
    keywords: [
        __( 'Single Event', 'the-events-calendar' ),
        __( 'The Events Calendar', 'the-events-calendar' ),
    ],
    edit: ( props ) => {
        const { className, ...blockProps } = useBlockProps();

        return (
            <div className={ `${ className } ${ props.className }` } { ...blockProps }>
                <h3>{ __( 'Event Title', 'the-events-calendar' ) }</h3>
                <p>
                    <strong>
                    { __( 'EVENT DATE/TIME', 'the-events-calendar' ) }
                </strong>
                </p>

                <FauxLine style={{marginLeft:34, marginRight:'25%'}} />
                <FauxLine style={{marginRight:'25%'}} />
                <FauxLine style={{marginRight:'25%'}} />
                <button type={"button"} class="wp-block-button__link wp-element-button">Subscribe to Event</button>
            </div>
        );
    },
};