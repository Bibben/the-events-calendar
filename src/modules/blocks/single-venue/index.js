/**
 * External Dependencies
 */
const { __ } = wp.i18n;
const { useBlockProps } = wp.blockEditor;


/**
 * Small component to simplify some pseudo event blocks.
 *
 * @param width
 * @returns {JSX.Element}
 * @constructor
 */
const EventItem = ( { width = '40%' } ) => {
    return (
        <div style={ { width, height: 16, background: '#eee', margin: '18px 4px' } }/>
    )
}


/**
 * The Venue block used in Site Editor templates.
 */
export default {
    id: 'tec/single-venue',
    title: __( 'Single Venue', 'the-events-calendar' ),
    icon: 'list-alt',
    category: 'tribe-events',
    keywords: [
        __( 'Single Venue', 'the-events-calendar' ),
        __( 'The Events Calendar', 'the-events-calendar' ),
    ],
    edit: ( props ) => {
        const { className, ...blockProps } = useBlockProps();

        return (
            <div className={ `${ className } ${ props.className }` } { ...blockProps }>
                <h3>{ __( 'Venue Title', 'the-events-calendar' ) }</h3>
                <div style={ { float: 'left', width: '40%' } }>
                    <EventItem width={ '70%' }/>
                    <EventItem width={ '84%' }/>
                    <EventItem width={ '73%' }/>
                    <EventItem width={ '63%' }/>
                </div>
                <div style={ { float: 'left', width: '30%' } }>
                    <div style={ {
                        width: '70%',
                        maxWidth: 340,
                        height: 180,
                        background: '#eee',
                        margin: '18px 4px'
                    } }/>
                </div>
                <div style={{clear:'both', height:1}}></div>
            </div>
        );
    },
};