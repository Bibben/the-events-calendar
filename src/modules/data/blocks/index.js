/**
 * Internal dependencies
 */
import * as datetime from './datetime';
import './sagas';

export { datetime };
export { default } from './reducer';
