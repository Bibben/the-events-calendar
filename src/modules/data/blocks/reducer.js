/**
 * External dependencies
 */
import { combineReducers } from 'redux';

/**
 * Internal dependencies
 */
import datetime, { setInitialState as datetimeSetInitialState } from './datetime';
import organizers from './organizers';
import price from './price';
import website from './website';
import venue from './venue';
import classic from './classic';
import sharing from './sharing';

export const setInitialState = ( entityRecord ) => {
	datetimeSetInitialState( entityRecord );
};

export default combineReducers( {
	datetime,
	classic,
	venue,
	organizers,
	price,
	website,
	sharing,
} );
