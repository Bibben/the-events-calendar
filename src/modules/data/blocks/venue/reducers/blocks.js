/**
 * External dependencies
 */
import { combineReducers } from 'redux';

/**
 * Internal dependencies
 */
import * as types from './../types';

export const byId = ( state = {}, action ) => {
	switch ( action.type ) {
		case types.SET_VENUE:
			return {
				[ action.payload.id ]: action.payload.venue
			};
		case types.ADD_BLOCK_VENUE:
			return {
				...state,
				[ action.payload.id ]: action.payload.venue,
			};
		case types.REMOVE_BLOCK_VENUE:
			return Object.keys( state ).reduce( ( newState, id ) => {
				if ( id === action.payload.id ) {
					return newState;
				}

				return {
					...newState,
					[ id ]: state[ id ],
				};
			}, {} );
		default:
			return state;
	}
};

export const allIds = ( state = [], action ) => {
	switch ( action.type ) {
		case types.SET_VENUE:
			return [ action.payload.venue ];
		case types.ADD_BLOCK_VENUE:
			return [ ...state, action.payload.venue ];
		case types.REMOVE_BLOCK_VENUE:
			return state.filter( venue => venue !== action.payload.venue );
		default:
			return state;
	}
};

export const core = ( state = {}, action ) => {
	switch ( action.type ) {
		case types.SET_VENUE:
			return {
				...state,
				venue: action.payload.venue,
			};
		case types.SET_VENUE_MAP:
			return {
				...state,
				showMap: action.payload.showMap,
			};
		case types.SET_VENUE_MAP_LINK:
			return {
				...state,
				showMapLink: action.payload.showMapLink,
			};
		default:
			return state;
	}
}

export default combineReducers( {
	core,
	byId,
	allIds,
} );
