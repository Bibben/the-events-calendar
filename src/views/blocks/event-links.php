<?php
/**
 * Block: Event Links
 *
 * Override this template in your own theme by creating a file at:
 * [your-theme]/tribe/events/blocks/event-links.php
 *
 * See more documentation about our Blocks Editor templating system.
 *
 * @link http://evnt.is/1aiy
 *
 * @version 4.7
 *
 */

use Tribe\Events\Views\V2\iCalendar\Links\Link_Abstract;

// don't show on password protected posts
if ( post_password_required() ) {
	return;
}

$has_google_cal   = $this->attr( 'hasGoogleCalendar' );
$has_ical         = $this->attr( 'hasiCal' );
$has_outlook_365  = $this->attr( 'hasOutlook365' );
$has_outlook_live = $this->attr( 'hasOutlookLive' );


remove_filter( 'the_content', 'do_blocks', 9 );
$subscribe_links = empty( $this->get( ['subscribe_links'] ) ) ? false : $this->get( ['subscribe_links'] );

$should_render  = $subscribe_links && ( $has_google_cal || $has_ical || $has_outlook_365 || $has_outlook_live );

$items = [];

if ( $has_google_cal && $this->get( [ 'subscribe_links', 'gcal' ] ) instanceof Link_Abstract ) {
	$items[] = $this->get( [ 'subscribe_links', 'gcal' ] );
}

if ( $has_ical && $this->get( [ 'subscribe_links', 'ical' ] ) instanceof Link_Abstract ) {
	$items[] = $this->get( [ 'subscribe_links', 'ical' ] );
}

if ( $has_outlook_365 && $this->get( [ 'subscribe_links', 'outlook-365' ] ) instanceof Link_Abstract ) {
	$items[] = $this->get( [ 'subscribe_links', 'outlook-365' ] );
}

if ( $has_outlook_live && $this->get( [ 'subscribe_links', 'outlook-live' ] ) instanceof Link_Abstract ) {
	$items[] = $this->get( [ 'subscribe_links', 'outlook-live' ] );
}

if ( empty( $items ) ) {
	return;
}

remove_filter( 'the_content', 'do_blocks', 9 );
?>
<div class="tribe-block tribe-block__events-link">
	<div class="tribe-events tribe-common">
		<div class="tribe-events-c-subscribe-dropdown__container">
			<div class="tribe-events-c-subscribe-dropdown">
				<div class="tribe-common-c-btn-border tribe-events-c-subscribe-dropdown__button" tabindex="0">
					<?php $this->template( 'v2/components/icons/cal-export', [ 'classes' => [ 'tribe-events-c-subscribe-dropdown__export-icon' ] ] ); ?>
					<button class="tribe-events-c-subscribe-dropdown__button-text">
						<?php echo esc_html__( 'Add to calendar', 'the-events-calendar' ); ?>
					</button>
					<?php $this->template( 'v2/components/icons/caret-down', [ 'classes' => [ 'tribe-events-c-subscribe-dropdown__button-icon' ] ] ); ?>
				</div>
				<div class="tribe-events-c-subscribe-dropdown__content">
					<ul class="tribe-events-c-subscribe-dropdown__list" tabindex="0">
						<?php foreach ( $items as $item ) : ?>
							<li class="tribe-events-c-subscribe-dropdown__list-item">
								<a
									href="<?php echo esc_url( $item->get_uri( null ) ); ?>"
									class="tribe-events-c-subscribe-dropdown__list-item-link"
									tabindex="0"
									target="_blank"
									rel="noopener noreferrer nofollow"
								>
									<?php echo esc_html( $item->get_label( null ) ); ?>
								</a>
							</li>
						<?php endforeach; ?>
					</ul>
				</div>
			</div>
		</div>
	</div>
</div>

<?php add_filter( 'the_content', 'do_blocks', 9 );
