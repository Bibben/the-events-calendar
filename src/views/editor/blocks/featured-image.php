<?php
/**
 * Block: Featured Image
 *
 * Override this template in your own theme by creating a file at:
 * [your-theme]/tribe/events/editor/blocks/featured-image.php
 *
 * See more documentation about our Blocks Editor templating system.
 *
 * @link {INSERT_ARTCILE_LINK_HERE}
 *
 * @version TBD
 *
 */

$event_id = $this->get( 'post_id' );
?>
<?php echo tribe_event_featured_image( $event_id, 'full', false );