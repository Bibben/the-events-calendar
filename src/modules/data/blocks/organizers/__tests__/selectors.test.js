/**
 * Internal dependencies
 */
import { selectors } from '@moderntribe/events/data/blocks/organizers';

const state = {
	events: {
		blocks: {
			organizers: {
				blocks: {
					byId: {
						firstBlock: {
							organizer: 100,
						},
						secondBlock: {
							organizer: 101,
						},
					},
					allIds: [ 100, 101 ],
				},
				classic: [ 98, 99, 100 ],
			},
		},
	}
};

describe( '[STORE] - Organizers selectors', () => {
	it( 'Should return the classic organizers', () => {
		expect( selectors.getOrganizersInClassic( state ) ).toMatchSnapshot();
	} );

	it( 'Should return the organizer block', () => {
		expect( selectors.getOrganizerBlock( state, { clientId: 'firstBlock' } ) )
			.toMatchSnapshot();
		expect( selectors.getOrganizerBlock( state, { clientId: 'secondBlock' } ) )
			.toMatchSnapshot();
		expect( selectors.getOrganizerBlock( state, { clientId: 'thirdBlock' } ) )
			.toMatchSnapshot();
	} );

	it( 'Should return the organizer in a block', () => {
		expect( selectors.getOrganizerInBlock( state, { clientId: 'firstBlock' } ) )
			.toMatchSnapshot();
		expect( selectors.getOrganizerInBlock( state, { clientId: 'secondBlock' } ) )
			.toMatchSnapshot();
		expect( selectors.getOrganizerInBlock( state, { clientId: 'thirdBlock' } ) )
			.toMatchSnapshot();
	} );

	it( 'Should return the organizers in a block', () => {
		expect( selectors.getOrganizersInBlock( state ) ).toMatchSnapshot();
	} );

	it( 'Should return the mapped organizers', () => {
		expect( selectors.getMappedOrganizers( state ) ).toMatchSnapshot();
	} );
} );
