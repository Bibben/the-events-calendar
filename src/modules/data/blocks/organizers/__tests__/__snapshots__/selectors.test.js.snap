// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[STORE] - Organizers selectors Should return the classic organizers 1`] = `
Array [
  98,
  99,
  100,
]
`;

exports[`[STORE] - Organizers selectors Should return the mapped organizers 1`] = `
Array [
  Object {
    "block": false,
    "id": 98,
  },
  Object {
    "block": false,
    "id": 99,
  },
  Object {
    "block": true,
    "id": 100,
  },
]
`;

exports[`[STORE] - Organizers selectors Should return the organizer block 1`] = `
Object {
  "organizer": 100,
}
`;

exports[`[STORE] - Organizers selectors Should return the organizer block 2`] = `
Object {
  "organizer": 101,
}
`;

exports[`[STORE] - Organizers selectors Should return the organizer block 3`] = `undefined`;

exports[`[STORE] - Organizers selectors Should return the organizer in a block 1`] = `100`;

exports[`[STORE] - Organizers selectors Should return the organizer in a block 2`] = `101`;

exports[`[STORE] - Organizers selectors Should return the organizer in a block 3`] = `undefined`;

exports[`[STORE] - Organizers selectors Should return the organizers in a block 1`] = `
Array [
  100,
  101,
]
`;
